File: app/src/main/java/yangFenTuoZi/runner/plus/ui/fragment/BaseFragment.java
package yangFenTuoZi.runner.plus.ui.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.core.view.MenuProvider;
import androidx.fragment.app.Fragment;

import yangFenTuoZi.runner.plus.ui.activity.MainActivity;

public class BaseFragment extends Fragment {
    public MainActivity mContext;

    public void setupToolbar(Toolbar toolbar, View tipsView, int title) {
        setupToolbar(toolbar, tipsView, getString(title), -1);
    }

    public void setupToolbar(Toolbar toolbar, View tipsView, int title, int menu) {
        setupToolbar(toolbar, tipsView, getString(title), menu);
    }

    public void setupToolbar(Toolbar toolbar, View tipsView, String title, int menu) {
        toolbar.setTitle(title);
        toolbar.setTooltipText(title);
        if (tipsView != null) tipsView.setTooltipText(title);
        if (menu != -1) {
            toolbar.inflateMenu(menu);
            if (this instanceof MenuProvider self) {
                self.onPrepareMenu(toolbar.getMenu());
            }
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = (MainActivity) requireActivity();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent resultData) {
        super.onActivityResult(requestCode, resultCode, resultData);
    }

    @Override
    public void startActivityForResult(@NonNull Intent intent, int requestCode) {
        super.startActivityForResult(intent, requestCode);
    }

    @Override
    public void startActivityForResult(@NonNull Intent intent, int requestCode, @Nullable Bundle options) {
        super.startActivityForResult(intent, requestCode, options);
    }

    public void runOnUiThread(Runnable action) {
        mContext.runOnUiThread(action);
    }
}

File: app/src/main/java/yangfentuozi/runner/base/BaseFragment.kt
package yangfentuozi.runner.base

import android.os.Bundle
import androidx.fragment.app.Fragment
import yangfentuozi.runner.ui.activity.MainActivity

open class BaseFragment : Fragment() {
    protected lateinit var mContext: MainActivity
    val appBar get() = mContext.appBar
    val toolbar get() = mContext.toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = (activity as? MainActivity)
            ?: throw RuntimeException("父Activity非MainActivity")
    }

    override fun onStart() {
        super.onStart()
        mContext.toolbar.subtitle = null
    }

    fun runOnUiThread(action: Runnable) {
        mContext.runOnUiThread(action)
    }
}